src/Rings/FreeAssAlgIdeal.jl

```diff
@@ -17,18 +17,24 @@ Two-sided ideal of a free associative algebra with elements of type `T`.
 mutable struct FreeAssAlgIdeal{T} <: Ideal{T}
   gens::IdealGens{T}
   gb::IdealGens{T}
-
-  function FreeAssAlgIdeal(R::FreeAssAlgebra, g::Vector{T}) where T <: FreeAssAlgElem
+  deg_bound::Int
+  function FreeAssAlgIdeal(g::IdealGens{T}, check_groebner_basis::Bool=false) where T <: FreeAssAlgElem
     r = new{T}()
-    r.gens = IdealGens(R, g)
+    r.gens = g
+    @req !check_groebner_basis || base_ring(base_ring(g)) isa Field "Groebner basis check requires a field base ring"
+    if check_groebner_basis && base_ring(base_ring(g)) isa Field && is_groebner_basis(g)
```

Comment:
I don't really understand what this new arg `check_groebner_basis` is supposed to do?
Could you write some explanation about that or maybe just put that directly into a docstring for `ideal(::IdeaGens; check_groebner_basis::Bool)`?

Reply:
It is supposed to check if the generator is already a Groebner basis and then add it to the .gb immediately. But in retrospect, I do not think it helps that much, especially if it is hidden behind a kwarg. I remove it.